frontend/src/components/Header/Search.jsx

```diff
@@ -1,15 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
 import { FaSearch } from 'react-icons/fa';
+import { AiFillCloseCircle } from 'react-icons/ai';
+import { SiReactos } from 'react-icons/si';
+
+import { getDataAPI } from '../../utils';
+import { useToast } from '../../hooks';
+import SearchUserCard from './SearchUserCard';
 
 const Search = () => {
+  const [search, setSearch] = useState('');
+  const [users, setUsers] = useState([]);
+
+  const { showToast } = useToast();
+
+  const { auth } = useSelector((state) => state);
+
+  const [loading, setLoading] = useState(false);
+
+  const handleSearch = async (e) => {
+    setSearch(e.target.value.toLowerCase().replace(/ /g, ''));
+
+    try {
+      setLoading(true);
+
+      const res = await getDataAPI(`search?username=${search}`, auth.token);
```

Comment:
I think you can apply debounce function here to reduce the api requests. It will surely boost your search functionality.